File: ModuleSample/app/src/main/java/com/example/modulesample/MainActivity.java
package com.example.modulesample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ListView;

public class MainActivity extends AppCompatActivity {

    ListView listView;
    Button btnAdd;
    ListAdapter adapter;

    int count;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = findViewById(R.id.v_listView);
        btnAdd = findViewById(R.id.btn_add);

        adapter = new ListAdapter();
        listView.setAdapter(adapter);

        ItemData data1 = new ItemData("데이터 가공");
        ItemData data2 = new ItemData("샘플2");
        ItemData data3 = new ItemData("샘플3");
        ItemData data4 = new ItemData("샘플4");

        adapter.addItem(data1);
        adapter.addItem(data2);
        adapter.addItem(data3);
        adapter.addItem(data4);

//        btnAdd.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                count = count++;
//                ItemData itemData = new ItemData(count + "");
//
//            }
//        });

    }
}

File: MVVM/app/src/main/java/com/puresoftware/mvvm/MainActivity.kt
package com.puresoftware.mvvm

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.lifecycle.ViewModelProvider

// https://brunch.co.kr/@mystoryg/175
class MainActivity : AppCompatActivity() {

    private lateinit var viewModel: OrderViewModel
    private lateinit var addButton: Button
    private lateinit var delButton: Button
    private lateinit var americanoCountText: TextView
    private lateinit var totalPriceText: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // view id
        addButton = findViewById(R.id.americanoAddButton)
        delButton = findViewById(R.id.americanoDeleteButton)
        americanoCountText = findViewById<Button>(R.id.americanoCountText)
        totalPriceText = findViewById<Button>(R.id.totalPriceText)

        //
        viewModel = ViewModelProvider(this).get(OrderViewModel::class.java)
        setAmericanoQtyObserver()
        setTotalPriceObserver()
        setAddButtonClickListener()
        setDelButtonClickListener()
    }

    private fun setAmericanoQtyObserver() {
        viewModel.americanoQty.observe(this) {
            americanoCountText.text = it
        }
    }

    private fun setTotalPriceObserver() {
        viewModel.totalPrice.observe(this) {
            totalPriceText.text = it
        }
    }

    private fun setAddButtonClickListener() {
        addButton.setOnClickListener {
            viewModel.addAmericano()
        }
    }

    private fun setDelButtonClickListener() {
        delButton.setOnClickListener {
            viewModel.delAmericano()
        }
    }
}